### ISSUE TITLE
Calculate compressed size

### ISSUE THREAD

**Issue by Claybird** [Claybird/lhaforge#24]
libarchive seems not to provide a way to know size of a compressed file entry. However, there could be some way to calculate from ftell or something similar.

- archive_read_open takes open/read/close callbacks; could use ftell to get original file position
- archive_position sounds to provide the same function in more elegant way

**Comment by Claybird** [comment]
archive_position_* is deprecated.

**Comment by Claybird** [comment]
I found a comment that
> [The “compressed size” is not exposed by libarchive. ](https://groups.google.com/g/libarchive-discuss/c/2x6ibwE3FL4?pli=1)

I tried to find an alternative way, but found it almost impossible. 

- archive_filter_bytes(_arc, -1) seems to return compressed file position, so subtracting the results should have been the compressed file size. Unfortunately, the result was somehow slightly different. Calculated size is larger than the expected value by about 60 bytes; the excess bytes varies between entries.

- Combination of archive_read_open_FILE() and ftell() gives completely wrong result.
